frontends/web/src/components/dialog/dialog.tsx

```diff
@@ -156,12 +156,23 @@ class Dialog extends Component<Props, State> {
       clearTimeout(this.timerId);
     }
 
-    this.modal.current?.classList.remove(style.open);
     this.overlay.current.classList.remove(style.activeOverlay);
     this.overlay.current.classList.add(style.closingOverlay);
-    this.timerId = setTimeout(() => {
+    this.modal.current?.classList.remove(style.open);
+
+    const onTransitionEnd = () => {
       this.deactivateModal(fireOnCloseProp);
-    }, 300);
+      this.modal.current?.removeEventListener('transitionend', onTransitionEnd);
+    };
+
+    const hasTransition = parseFloat(window.getComputedStyle(this.modal.current).transitionDuration) > 0;
+    if (hasTransition) {
+      this.overlay.current.addEventListener('transitionend', onTransitionEnd);
+      // fallback in-case the 'transitionend' event didn't fire
+      this.timerId = setTimeout(onTransitionEnd, 300);
```

Comment:
at least on mobile the onClose callback is called multiple times.

I haven't tested yet, but how about we fully depend on 'transitionend' and also use it on dekstop with `0s` transition WDYT?

Reply:
> how about we fully depend on 'transitionend' 

thinking about it a bit more: there is also the reduce-animation OS settings, which we had to test if it fires.